src/components/menu/Menu.tsx

```diff
@@ -199,7 +201,7 @@ class Menu extends React.Component<MenuProps> {
         const { menuItemEl }: { menuItemEl?: HTMLElement | null } =
             event.target instanceof Node ? this.getMenuItemElFromEventTarget(event.target) : {};
 
-        if (!menuItemEl) {
+        if (!menuItemEl || (event.target as Element).className.includes('submenu-target')) {
```

Comment:
Can you elaborate on what this checking? Checking the `className` seems fragile to me, but wondering if it's okay here if we are not going to change it.

Reply:
This is to not close the menu when the menu element is the triggering submenu element. This is not okay on mobile since you don't have hover events, just touch events.

Reply:
I mean, I think there is probably a better solution here...I'll take a look at this again.

Reply:
I ended up moving this logic to `SubmenuItem`. Basically if the submenu triggering element is being clicked, prevent propagation/default